src/main/java/org/minions/devfund/marylin/Squarelotron.java

```diff
@@ -0,0 +1,273 @@
+            for (int i = 1; i < size - 1; i++) {
+                for (int j = 1; j < size - 1; j++) {
+                    resultGrid[i][j] = this.squarelotron[i][j];
+                }
+            }
+        }
+        return resultGrid;
+    }
+
+
+    /**
+     * flipExternalRing grid.
+     *
+     * @param matrix grid.
+     * @return a grid.
+     */
+    private int[][] flipExternalRing(final int[][] matrix) {
+        int[][] resultGrid = duplicateGrid(matrix);
+        int tempJ = 0;
+        for (int i = 1; i < this.size; i++) {
+            resultGrid[i][tempJ] = matrix[tempJ][i];
+            resultGrid[tempJ][i] = matrix[i][tempJ];
+        }
+
+        int tempI = this.size - 1;
+        for (int j = 1; j < this.size - 1; j++) {
+            resultGrid[tempI][j] = matrix[j][tempI];
+            resultGrid[j][tempI] = matrix[tempI][j];
+        }
+        return resultGrid;
+    }
+
+    /**
+     * flipInternalRing a grid.
+     *
+     * @param matrix grid.
+     * @return a grid.
+     */
+    private int[][] flipInternalRing(final int[][] matrix) {
+        int[][] resultGrid = duplicateGrid(matrix);
+        int tempJ = 1;
+        for (int i = 2; i < size - 1; i++) {
+            resultGrid[i][tempJ] = matrix[tempJ][i];
+            resultGrid[tempJ][i] = matrix[i][tempJ];
+        }
+
+        int tempI = size - 2;
+        for (int j = 2; j <= size - 2; j++) {
+            resultGrid[tempI][j] = matrix[j][tempI];
+            resultGrid[j][tempI] = matrix[tempI][j];
+        }
+        return resultGrid;
+    }
+
+    /**
+     * mainDiagonalFlip.
+     *
+     * @param ring ring number.
+     * @return a Squarelotron.
+     */
+    Squarelotron mainDiagonalFlip(int ring) {
+        int[][] resultGrid = mainDiagonalFlipHelper(ring);
+        return new Squarelotron(resultGrid);
+    }
+
+    /**
+     * mainDiagonalFlipHelper.
+     *
+     * @param ring ring number.
+     * @return a grid.
+     */
+    private int[][] mainDiagonalFlipHelper(int ring) {
+        int[][] resultGrid;
+        // returns the same squarelotron if the ring number is equal to the number of rings.
+        if (Math.abs(this.size) % 2 == 1 && ring == this.numberOfRings) {
+            return this.squarelotron;
+        }
+        if (ring != 1) {
+            resultGrid = flipInternalRing(this.squarelotron);
+        } else {
+            resultGrid = flipExternalRing(this.squarelotron);
+        }
+        return resultGrid;
+    }
+
+    /**
+     * rotateEast.
+     *
+     * @param numberOfTurns number of turns.
+     */
+    private void rotateEast(int numberOfTurns) {
+        while (numberOfTurns > 0) {
+            for (int i = 1; i <= numberOfRings; i++) {
+                squarelotron = upsideDownFlipHelper(i);
+            }
+            for (int i = 1; i <= numberOfRings; i++) {
+                squarelotron = mainDiagonalFlipHelper(i);
+            }
+            numberOfTurns--;
+        }
```

Comment:
This may be refactored like this.
```java
    private void rotateEast(int numberOfTurns) {
        while (numberOfTurns-- > 0) {
            for (int i = 1; i <= numberOfRings; i++) {
                squarelotron = upsideDownFlipHelper(i);
                squarelotron = mainDiagonalFlipHelper(i);
            }
        }
    }
```

Reply:
Done